core/src/main/java/io/questdb/cairo/wal/WalEventReader.java

```diff
@@ -128,4 +140,14 @@ public WalEventCursor of(Path path, long segmentTxn) {
             path.trimTo(trimTo);
         }
     }
+
+    /**
+     * Read a non-negative long at the specified offset, or return -1 if out of bounds.
+     */
+    private static long readNonNegativeLong(MemoryCMR eventIndexMem, long offset) {
+        if ((offset < 0) || (eventIndexMem.addressOf(offset + Long.BYTES) < 0)) {
```

Comment:
too many unnecessary checks:

- offset in two out of 3 cases is already checked for negative value
- addressOf() is designed to throw assert error if address is unmapped, we don't return negative values randomly anymore
- if we read "maxTxn" as negative, we should add an explicit validation rather than defaulting to a generic error

Reply:
I was simply trying to port the existing logic as closely as possible.
Let me roll in these improvements as well.

Reply:
I've reworked the checks.